### PR TITLE
fix: patch to refetch relevant data for historic vaults values in charts

### PR DESCRIPTION
**Pull request by NamelsKing**

### CODE REVIEW

File: src/patches/vault-historic-data-refetch.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          Unable to query cached chart, for vault ${vaultChartDataKey} may simply not exist.
+          ${e}
+        `);
+        cachedChart = undefined;
+      }
+
+      if (!cachedChart) continue;
+
+      for (const snapshot of cachedChart.data) {
+        if (!snapshot.strategy || (snapshot.strategy && snapshot.strategy.address.startsWith('0x0000'))) {
```
Comment (axejintao) [inline src/patches/vault-historic-data-refetch.ts:85]: we can check it against `ethers.constants.ZeroAddress` in the case that somehow someone deployed / deploys a vanity address however, for this patch not really required so its a nit

File: src/patches/vault-historic-data-refetch.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      continue;
+    }
+
+    let cachedChart: ChartDataBlob<HistoricVaultSnapshotModel> | undefined;
+    const vaultEntityId = getVaultEntityId(chain, vault);
+
+    let strategyInfo: VaultStrategy | undefined;
+
+    try {
+      strategyInfo = await getStrategyInfo(chain, vault);
```
Comment (axejintao) [inline src/patches/vault-historic-data-refetch.ts:58]: hmm, should we utilize the `blockTag` on entries? maybe it is ok not to as it does not change very often for this purpose (none changed during the issue period)

**Review by axejintao — COMMENTED** [review]
no blockers, some nits - lmk your thoughts can approve and run later

File: src/patches/vault-historic-data-refetch.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      } catch (e) {
+        console.debug(`
+          Patch:VaultSnapshotRefetch - 
+          Unable to query cached chart, for vault ${vaultChartDataKey} may simply not exist.
+          ${e}
+        `);
+        cachedChart = undefined;
+      }
+
+      if (!cachedChart) continue;
```
Comment (mrbasado) [inline src/patches/vault-historic-data-refetch.ts:82]: on 1->n of this loop the cacheChart never gets reset to undefined for this if check

Comment (mrbasado) [reply src/patches/vault-historic-data-refetch.ts:82]: i should rephrase, i see the set in the catch but not sure if that is the only case, if it is then the continue should be inside the catch{} as well

Comment (NamelsKing) [reply src/patches/vault-historic-data-refetch.ts:82]: it will, in case of error

File: src/patches/vault-historic-data-refetch.ts (hunk 2 above)
Comment (NamelsKing) [reply src/patches/vault-historic-data-refetch.ts:58]: ohh shoot, good point ty

File: src/patches/vault-historic-data-refetch.ts (hunk 3 above)
Comment (NamelsKing) [reply src/patches/vault-historic-data-refetch.ts:77]: doesnt rly matter, but ye, can be in catch block

File: src/patches/vault-historic-data-refetch.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      }
+
+      try {
+        await mapper.put(cachedChart);
+      } catch (err) {
+        console.error(`Patch:VaultSnapshotRefetch - Failed to save 
+        vault chart data for ${vaultChartDataKey}. ${err}`);
+      }
+
+      console.log(`Patch:VaultSnapshotRefetch - Chart ${cachedChart.id} updated.`);
```
Comment (mrbasado) [inline src/patches/vault-historic-data-refetch.ts:103]: this still prints after failure

File: src/patches/vault-historic-data-refetch.ts (hunk 3 above)
Comment (mrbasado) [reply src/patches/vault-historic-data-refetch.ts:77]: i would change it for readability

File: src/patches/vault-historic-data-refetch.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+              Patch:VaultSnapshotRefetch - Failed to fetch strategyInfo
+              for vaultChartDataKey ${vaultChartDataKey}
+            ${err}`);
+
+            continue;
+          }
+
+          snapshot.strategy = strategyInfo;
+        }
+      }
```
Comment (mrbasado) [inline src/patches/vault-historic-data-refetch.ts:94]: when the next for loop starts cachedChart will start with this old value no?

**Review by mrbasado — APPROVED** [review]

File: src/patches/vault-historic-data-refetch.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+ * Data to be updated:
+ * - `strategy` field
+ *
+ * https://github.com/Badger-Finance/badger-api/issues/1539
+ */
+
+async function vaultHistoricDataRefetch() {
+  console.log('Patch:VaultSnapshotRefetch - Started');
+
+  getEnvVar('STAGE');
```
Comment (axejintao) [inline src/patches/vault-historic-data-refetch.ts:36]: does this need to be stored or used?

**Review by axejintao — COMMENTED** [review]
one q, otherwise lgtm

File: src/patches/vault-historic-data-refetch.ts (hunk 6 above)
Comment (NamelsKing) [reply src/patches/vault-historic-data-refetch.ts:36]: use more like protector, need more for ddb setup

Comment (NamelsKing) [reply src/patches/vault-historic-data-refetch.ts:36]: so no

**Review by axejintao — APPROVED** [review]

**Pull request merged**